sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/WriteToDataSourceV2Exec.scala

```diff
@@ -196,11 +204,19 @@ case class AtomicReplaceTableAsSelectExec(
     partitioning: Seq[Transform],
     plan: LogicalPlan,
     query: SparkPlan,
-    properties: Map[String, String],
+    tableSpec: TableSpec,
     writeOptions: CaseInsensitiveStringMap,
     orCreate: Boolean,
     invalidateCache: (TableCatalog, Table, Identifier) => Unit) extends TableWriteExecHelper {
 
+  val properties = {
+    val props = CatalogV2Util.convertTableProperties(
```

Comment:
nit: should the util function take `TableSpec` directly? and also fill the ownership property. Then we can simplify the code here and other places.

Reply:
Fixed. Thanks!